src/CourierClient.php

```diff
@@ -844,7 +844,125 @@ public function postNotificationDraftVariations(string $id, array $blocks, array
+     *
+     * @param string $id
+     * @param string $blockId
+     * @param array $locales
+     * @return object
+     * @throws CourierRequestException
+     * @throws \Psr\Http\Client\ClientExceptionInterface
+     */
+    public function putNotificationDraftBlockLocales(string $id, string $blockId, array $locales): object
+    {
+        return $this->doRequest(
+            $this->buildRequest("put", "notifications/" . $id . "/draft/blocks/". $blockId ."/locales", $locales)
+        );
+    }
+
+    /**
+     * Put notification channel locales
+     *
+     * @param string $id
+     * @param string $channelId
+     * @param array $locales
+     * @return object
+     * @throws CourierRequestException
+     * @throws \Psr\Http\Client\ClientExceptionInterface
+     */
+    public function putNotificationChannelLocales(string $id, string $channelId, array $locales): object
+    {
+        return $this->doRequest(
+            $this->buildRequest("put", "notifications/" . $id . "/channels/". $channelId ."/locales", $locales)
+        );
+    }
+
+    /**
+     * Put notification draft channel locales
+     *
+     * @param string $id
+     * @param string $channelId
+     * @param array $locales
+     * @return object
+     * @throws CourierRequestException
+     * @throws \Psr\Http\Client\ClientExceptionInterface
+     */
+    public function putNotificationDraftChannelLocales(string $id, string $channelId, array $locales): object
+    {
+        return $this->doRequest(
+            $this->buildRequest("put", "notifications/" . $id . "/draft/channels/". $channelId ."/locales", $locales)
+        );
+    }
+
+    /**
+     * Put notification locale
+     *
+     * @param string $id
+     * @param string $localeId
+     * @param array $blocks
+     * @param array|null $channels
+     * @return object
+     * @throws CourierRequestException
+     * @throws \Psr\Http\Client\ClientExceptionInterface
+     */
+    public function putNotificationLocale(string $id, string $localeId, array $blocks, array $channels = NULL): object
+    {
+        $params = array(
+            'blocks' => $blocks,
+            'channels' => $channels
+        );
+
+        $params = array_filter($params);
+
+        return $this->doRequest(
+            $this->buildRequest("put", "notifications/" . $id . "/locales/". $localeId, $params)
+        );
+    }
+
+    /**
+     * Put notification draft locale
+     *
+     * @param string $id
+     * @param string $localeId
+     * @param array $blocks
+     * @param array|null $channels
+     * @return object
+     * @throws CourierRequestException
+     * @throws \Psr\Http\Client\ClientExceptionInterface
+     */
+    public function putNotificationDraftLocale(string $id, string $localeId, array $blocks, array $channels = NULL): object
+    {
+        $params = array(
+            'blocks' => $blocks,
+            'channels' => $channels
+        );
+
+        $params = array_filter($params);
```

Comment:
same here, i would maybe just do the following

```suggestion

        $params = array(
            'blocks' => $blocks,
        );

        if ( $channels !== NULL ) $params['channels'] = $channels;
```